docs/guide/blog/comment-blog.md

```diff
@@ -0,0 +1,548 @@
+	store.Set(byteKey, appendedValue)
+    // Update the comment count
+	k.SetCommentCount(ctx, count+1)
+	return count
+}
+```
+
+By following these steps, you have implemented all of the code required to create new comments and store them on-chain. Now, when a transaction that contains a message of type `MsgCreateComment` is broadcast, the message is routed to your blog module.
+
+- `x/blog/handler.go` calls `k.CreateComment` which in turn calls `AppendComment`.
+- `AppendComment` gets the number of comments from the store, adds a comment using the count as an ID, increments the count, and returns the ID.
+
+
+## Display Posts
+
+```bash
+starport scaffold query comments --response title,body,blogID
+```
+
+Very similar to previous blog tutorial, we will make changes to `proto/blog/query.proto`
+
+In the `proto/blog/query.proto` file:
+
+```go
+// Import the Comment message
+import "blog/comment.proto";
+
+message QueryCommentsRequest {
+    // Adding pagination to request
+  cosmos.base.query.v1beta1.PageRequest pagination = 1;
+}
+//...
+message QueryCommentsResponse {
+    // Returning a list of comments
+  repeated Comment Comment = 1;
+    // Adding pagination to response
+  cosmos.base.query.v1beta1.PageResponse pagination = 2;
+}
+```
+
+After the types are defined in proto files, you can implement post querying logic. In `grpc_query_comments.go`:
+
+```go
+package keeper
+
+import (
+	"context"
+
+	"github.com/cosmonaut/blog/x/blog/types"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (k Keeper) Comments(c context.Context, req *types.QueryCommentsRequest) (*types.QueryCommentsResponse, error) {
+	// Throw an error if request is nil
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+	// Define a variable that will store a list of posts
+	var comments []*types.Comment
+	// Get context with the information about the environment
+	ctx := sdk.UnwrapSDKContext(c)
+	// Get the key-value module store using the store key (in our case store key is "chain")
+	store := ctx.KVStore(k.storeKey)
+	// Get the part of the store that keeps posts (using post key, which is "Post-value-")
+	commentStore := prefix.NewStore(store, []byte(types.CommentKey))
+	// Paginate the posts store based on PageRequest
+	pageRes, err := query.Paginate(commentStore, req.Pagination, func(key []byte, value []byte) error {
+		var comment types.Comment
+		if err := k.cdc.Unmarshal(value, &comment); err != nil {
+			return err
+		}
+		comments = append(comments, &comment)
+		return nil
+	})
+	// Throw an error if pagination failed
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	// Return a struct containing a list of posts and pagination info
+	return &types.QueryCommentsResponse{Comment: comments, Pagination: pageRes}, nil
```

Comment:
The whole point of this tutorial is to write a query that returns a blog post by ID with associated comments.